broker/src/main/java/io/camunda/zeebe/broker/transport/commandapi/CommandApiRequestHandler.java

```diff
@@ -8,161 +8,112 @@
-import java.util.Map;
-import java.util.Queue;
-import java.util.function.Consumer;
-import org.agrona.DirectBuffer;
+import io.camunda.zeebe.util.Either;
 import org.agrona.collections.Int2ObjectHashMap;
-import org.agrona.concurrent.ManyToOneConcurrentLinkedQueue;
 import org.slf4j.Logger;
 
-final class CommandApiRequestHandler implements RequestHandler {
+final class CommandApiRequestHandler
+    extends ApiRequestHandler<CommandApiRequestReader, CommandApiResponseWriter> {
   private static final Logger LOG = Loggers.TRANSPORT_LOGGER;
 
-  private final MessageHeaderDecoder messageHeaderDecoder = new MessageHeaderDecoder();
-  private final ExecuteCommandRequestDecoder executeCommandRequestDecoder =
-      new ExecuteCommandRequestDecoder();
-  private final Queue<Runnable> cmdQueue = new ManyToOneConcurrentLinkedQueue<>();
-  private final Consumer<Runnable> cmdConsumer = Runnable::run;
-
   private final Int2ObjectHashMap<LogStreamRecordWriter> leadingStreams = new Int2ObjectHashMap<>();
   private final Int2ObjectHashMap<RequestLimiter<Intent>> partitionLimiters =
       new Int2ObjectHashMap<>();
-  private final RecordMetadata eventMetadata = new RecordMetadata();
-
-  private final ErrorResponseWriter errorResponseWriter = new ErrorResponseWriter();
-
-  private final Map<ValueType, UnpackedObject> recordsByType = new EnumMap<>(ValueType.class);
-  private final BackpressureMetrics metrics;
+  private final BackpressureMetrics metrics = new BackpressureMetrics();
   private boolean isDiskSpaceAvailable = true;
 
   CommandApiRequestHandler() {
-    metrics = new BackpressureMetrics();
-    initEventTypeMap();
+    super(new CommandApiRequestReader(), new CommandApiResponseWriter());
   }
 
-  private void initEventTypeMap() {
-    recordsByType.put(ValueType.DEPLOYMENT, new DeploymentRecord());
-    recordsByType.put(ValueType.JOB, new JobRecord());
-    recordsByType.put(ValueType.PROCESS_INSTANCE, new ProcessInstanceRecord());
-    recordsByType.put(ValueType.MESSAGE, new MessageRecord());
-    recordsByType.put(ValueType.JOB_BATCH, new JobBatchRecord());
-    recordsByType.put(ValueType.INCIDENT, new IncidentRecord());
-    recordsByType.put(ValueType.VARIABLE_DOCUMENT, new VariableDocumentRecord());
-    recordsByType.put(ValueType.PROCESS_INSTANCE_CREATION, new ProcessInstanceCreationRecord());
+  @Override
+  protected Either<ErrorResponseWriter, CommandApiResponseWriter> handle(
+      final int partitionId,
+      final long requestId,
+      final CommandApiRequestReader requestReader,
+      final CommandApiResponseWriter responseWriter,
+      final ErrorResponseWriter errorWriter) {
+    final var templateId = requestReader.getMessageDecoder().sbeTemplateId();
+
+    if (templateId == ExecuteCommandRequestDecoder.TEMPLATE_ID) {
+      return handleExecuteCommandRequest(
+          partitionId, requestId, requestReader, responseWriter, errorWriter);
+    } else {
+      return Either.left(errorWriter);
```

Comment:
:question:  Here isn't errorWriter is empty? We haven't added an error.

Reply:
You are right. I thought that was also the behaviour before my changes but it's not. I'll re-add this and try and write a test for it.

Reply:
I've fixed this but I required a bit more changes than I would have liked: https://github.com/camunda-cloud/zeebe/pull/8257/commits/f1a94fd1c8b1b434209b143c7ff04221b9ea55f2